hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/services/events/HmppsDomainEventEmitter.kt

```diff
@@ -91,36 +80,23 @@ class HmppsDomainEventEmitter(
     val prisonerUpdatedEvent = PrisonerUpdatedEvent(offenderNo, differences.keys.toList().sorted())
     val event = PrisonerUpdatedDomainEvent(prisonerUpdatedEvent, Instant.now(clock), diffProperties.host)
     if (red) {
-      event.publish {
-        log.error("Failed to send event $UPDATED_EVENT_TYPE for offenderNo=$offenderNo, differences=$differences. Event will be retried")
-        throw it
-      }
+      event.publish { logError(event, it, differences) }
```

Comment:
The main difference is that now there is no point in re-throwing the exception as the feature of it causing a db rollback no longer applies. So the red index update is not rolled back and a retry would just result in a no-op and therefore no event generation. 
So the strategy now is to alert on the EVENTS_SEND_FAILURE and intervene manually. This is ok as the failure is extremely rare (no occurrences in 2025 anyway)